### PR TITLE
Add missing CHECK_PARSER_STATUS

### PR DESCRIPTION
**Pull request by cyyever**
When reviewing the parser code, I found some invocations without checking the return value.

### CODE REVIEW

**Review by justinchuby — APPROVED** [review]

File: onnx/defs/parser.cc [hunk 1]
```diff
@@ -729,19 +729,19 @@ Status OnnxParser::Parse(NodeProto& node) {
   MATCH('=');
   std::string domain("");
   std::string id;
-  ParseIdentifier(id);
+  (void)(ParseIdentifier(id));
```
Comment (justinchuby) [inline onnx/defs/parser.cc:732]: Could you comment on this? Why is this needed?

Comment (gramalingam) [reply onnx/defs/parser.cc:732]: Yes, I think this should also be a CHECK_PARSER_STATUS: I don't think an empty identifier (which is the only possible error here) is allowed here.

Comment (cyyever) [reply onnx/defs/parser.cc:732]: One test specifies an empty id. This change is to make it green.

Comment (gramalingam) [reply onnx/defs/parser.cc]: Ok, thanks! I would say that the test-case is buggy, for an omitted domain, the "." is omitted too: we are supposed to use `MatMul` or `ai.onnx.MatMul` or `ai.onnx.ml.SVMClassifier`, not `.MatMul` ... but that can be fixed in a separate PR, this looks good to me now.

**Review by gramalingam — APPROVED** [review]

File: onnx/defs/parser.cc (hunk 1 above)
Comment (cyyever) [reply onnx/defs/parser.cc]: @gramalingam I'm not familiar with ONNX domain format. Therefore, need your PR help if further fixes are required.
